runtime/src/validate_block/implementation.rs

```diff
@@ -220,6 +233,24 @@ impl<B: BlockT> Storage for WitnessStorage<B> {
 			self.overlay.insert(key, None);
 		}
 	}
+
+	fn storage_append(&mut self, key: &[u8], value: Vec<u8>) {
+		let key_vec = key.to_vec();
+		let mut value_vec = Vec::with_capacity(1);
+		value_vec.push(EncodeOpaqueValue(value));
+
+		if let Some(Some(item)) = self.overlay.remove(&key_vec) {
```

Comment:
You want to use the `entry` api here.

Reply:
I don't think I want this time. I'm doing a remove to extract the value to transform it and re-insert it in the hashmap. I don't need a mutable reference here

Reply:
You want it ;) `sp_std::mem::take` is your friend.

Reply:
Probably fixed in here: https://github.com/paritytech/cumulus/pull/92/files#r422010526